examples/asa/assets/5-contract-owned-asa.teal

```diff
@@ -0,0 +1,51 @@
+#pragma version 2
+// Check if Transaction type is axfer(asset transfer), asset amount is <= 100 and 
+// only owner can transfer ASA out from contract
+// Also check for asset creation transaction.
+txn TypeEnum
+int 4
```

Comment:
Is it possilbe to use contants / enums instead of numbers? (4 above, other numbers below)
If not then please add a comment to what 4 is (and below to what 3 is).

Also, maybe we should limit the Creation of ASA? Is it possible to check that it also has to be signed by the owner? 

+ Please update README and comment in the top of this smart contract.

Reply:
Yes, we can use `axfer` and `acfg` , adding a comment,

what do you mean by limit creation ? limit per account ?

I tried that by delegating the signature with owner's sign, but we cannot do that, if we do that we can only send ASA from owner's account by validating lsig, and not from lsig account. when i tried to do later one i got logic signature validation failed

OK

Reply:
> what do you mean by limit creation ? limit per account ?

Yes - that only one account can create contract ASA.

Reply:
@amityadav0  - try to ask on Discord Algorand channel how to do it.

Reply:
yes asked on discord and updated example.
but for co-signed we need to group transaction, added task for it